lib/WebService/PayPal/PaymentsAdvanced/Response/SecureToken.pm

```diff
@@ -82,6 +100,56 @@ sub _build_hosted_form_uri {
     );
 }
 
+# Make an HTTP request to the given URI.
+#
+# Return the response if the request is successful. If the request fails with a
+# 5xx error, retry. We retry up to the configured number of times. On
+# encountering a non-5xx and non-success response, return the response
+# immediately.
+#
+# Throw an error if we exhaust our retries.
+sub _make_http_request_with_retries {
+    my $self = shift;
+    my $uri  = shift;
+
+    my $res;
+
+    # +1 so we always try at least once.
+    ## no critic (ControlStructures::ProhibitCStyleForLoops)
+    for (
+        my $attempts_left = $self->retry_attempts + 1;
+        $attempts_left > 0;
+        $attempts_left--
+        ) {
```

Comment:
Any reason to not just do `for my $attempt (1..$self->retry_attempts+1)` here and adjust the `last` check appropriately? The C-style for loop is a bit harder to parse and more prone to off-by-one errors.